### PR TITLE
docs: adds documentation for RPC headers.

### PR DESCRIPTION
**Pull request by jcchavezs**
This PR adds complementary documentation to https://github.com/grpc-ecosystem/grpc-health-probe/pull/72

Ping @ahmetb @drigz @farzadmf

### CODE REVIEW

**Comment by farzadmf** [comment]
Nice @jcchavezs , thank you for tagging :slightly_smiling_face:

File: README.md [hunk 1]
```diff
@@ -155,7 +155,8 @@ environment variable.
       $ grpc_health_probe -addr 127.0.0.1:10000 \
           -tls \
           -tls-ca-cert /path/to/testdata/ca.pem \
-          -tls-server-name=x.test.youtube.com
+          -tls-server-name=x.test.youtube.com \
+          -rpc-header=x-request-id:be59908d40f00be3573a50284c3863f1a37b8528
```
Comment (ahmetb) [inline README.md:159]: You can undo this part. This example is for routeguide/ app, the previous flag is actually used, but this won't be.

Comment (jcchavezs) [reply README.md:159]: My idea is to show an example in the readme. Shall we add another example just for rpc-header?

**Comment by jcchavezs** [comment]
Ping

File: README.md (hunk 1 above)
Comment (ahmetb) [reply README.md:159]: I think we can just change it to -rpc-header=foo:bar` (the example you provided doesn't resemble an example :))

Comment (ahmetb) [reply README.md:159]: I will fix the example myself. 😇

**Pull request merged**